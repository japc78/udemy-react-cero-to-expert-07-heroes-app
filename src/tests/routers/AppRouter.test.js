import React from 'react'
import { mount } from 'enzyme'
import { AppRouter } from '../../routers/AppRouter'
import { AuthContext } from '../../auth/AuthContext'

describe('Test on <AppRouter />', () => {

    test('should shown the login if it is not authenticated', () => {

        // Contexto inicial para la prueba.
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: false
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        // console.log(wrapper.html());
        expect(wrapper).toMatchSnapshot();
    });

    test('should shown Marvel component if it is authenticated', () => {
        // Contexto inicial para la prueba.
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Bitcero'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        // console.log(wrapper.html());
        expect(wrapper.find('.navbar').exists()).toBe(true);
    });

})
