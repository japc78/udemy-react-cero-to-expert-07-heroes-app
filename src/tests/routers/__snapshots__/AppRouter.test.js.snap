// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter /> should shown the login if it is not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login Screen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      login
    </button>
  </div>
</div>
`;
